pkg/cmd/build/log_filter.go

```diff
@@ -0,0 +1,79 @@
+package build
+
+import (
+	"fmt"
+	"github.com/moby/buildkit/client"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"regexp"
+)
+
+type ConditionFunc func(vertex *client.Vertex, ss *client.SolveStatus) bool
+type TransformerFunc func(vertex *client.Vertex, ss *client.SolveStatus, progress string)
+
+type Rule struct {
+	condition   ConditionFunc
+	transformer TransformerFunc
+}
+
+type BuildKitLogsFilter struct {
+	rules []Rule
+}
+
+func NewBuildKitLogsFilter(rules []Rule) *BuildKitLogsFilter {
+	return &BuildKitLogsFilter{
+		rules: rules,
+	}
+}
+
+func (lf *BuildKitLogsFilter) Run(ss *client.SolveStatus, progress string) {
+	for _, vertex := range ss.Vertexes {
+		for _, rule := range lf.rules {
+			if rule.condition(vertex, ss) {
+				rule.transformer(vertex, ss, progress)
+			}
+		}
+	}
+}
+
+// BuildKitMissingCacheCondition checks if the log contains a missing cache error
+var BuildKitMissingCacheCondition ConditionFunc = func(vertex *client.Vertex, ss *client.SolveStatus) bool {
```

Comment:
```suggestion
var BuildKitMissingCacheCondition ConditionFunc = func(vertex *client.Vertex) bool {
```
do we need that param?

Reply:
actually not, it was used at the beginnign and then removed, good shout! Removed in https://github.com/okteto/okteto/pull/3541/commits/05c6061c2d53e4b0d54addfd2555bce7ef178c0e